### PR TITLE
Fix stale audience in navigation stack

### PR DESCRIPTION
**Pull request by bfollington**
Relies on https://github.com/subconsciousnetwork/subconscious/pull/837
Fixes https://github.com/subconsciousnetwork/subconscious/issues/798

Prefer updated address to stale address after changing audience.

Interestingly, this _also_ fixes https://github.com/subconsciousnetwork/subconscious/issues/583

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Detail/MemoEditorDetail.swift [hunk 1]
```diff
@@ -987,7 +987,7 @@ struct MemoEditorDetailModel: ModelProtocol {
         info: MemoEditorDetailDescription
     ) -> Update<MemoEditorDetailModel> {
         // No address? This is a draft.
-        guard let address = state.address ?? info.address else {
+        guard let address = info.address ?? state.address else {
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Detail/MemoEditorDetail.swift:990]: Reasoning about what this inversion is doing is tricky. Can you explain what's happening?

**Review by gordonbrander — APPROVED** [review]
LGTM, but I have a question before merging.

File: xcode/Subconscious/Shared/Components/Detail/MemoEditorDetail.swift (hunk 1 above)
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Detail/MemoEditorDetail.swift:990]: When we rewrite the navigation stack after a rename we rewrite the list of `details` to point to the new address (with updated audience). When we pop the navigation stack we dispatch `send(.appear(description))` in `onAppear {}` which contains the _updated_ address. This is our chance to reload the data and refresh the view.

The bug was that we were throwing the _new_ address away (with updated audience) and retaining the address stored in `state`, which no longer resolves. 

TL;DR, in `.appear`'s handler, `state` is stale and `info` is fresh.

Comment (gordonbrander) [reply xcode/Subconscious/Shared/Components/Detail/MemoEditorDetail.swift:990]: That scans. Let's get this merged 🙏
